src/track/beatmap.cpp

```diff
@@ -481,15 +481,14 @@ std::unique_ptr<BeatIterator> BeatMap::findBeats(
         return std::unique_ptr<BeatIterator>();
     }
 
-    Beat startBeat = beatFromFramePos(startPosition);
-    Beat endBeat = beatFromFramePos(endPosition);
+    Beat startBeat = beatFromFramePos(startPosition.toUpperFrameBoundary());
```

Comment:
What is the rational to shrink the range here compared to the original version? 
I think a comment would be helpful.

Reply:
It this whole function unused? Delete!

Reply:
Oh, sorry, it is used in WaveformRenderBeat::draw() to find the visible beats in a time range. So it is still required.

Reply:
> What is the rational to shrink the range here compared to the original version? 
> I think a comment would be helpful.
>  

Beats can only appear a full frame positions. 
Let's say a beat is at pos 100. If I say "Give me all beats between 100.5 and 200", it should give me the beat at 100.

Reply:
https://github.com/mixxxdj/mixxx/pull/4095/commits/2d47336b7589c7524afd797f6945fd7d87faaa76